src/wp-includes/class-wp-query.php

```diff
@@ -2501,12 +2502,42 @@ public function get_posts() {
 			if ( isset( $query_vars['orderby'] ) && ( is_array( $query_vars['orderby'] ) || false === $query_vars['orderby'] ) ) {
 				$orderby = '';
 			} else {
-				$orderby = "{$wpdb->posts}.post_date " . $query_vars['order'];
+				/*
+				 * Ensure deterministic ordering to prevent duplicate records across pages.
+				 * When multiple posts have the same value for a field, add ID as secondary sort to guarantee consistent ordering.
+				 * Note: this is to circumvent a bug that is currently being tracked in https://core.trac.wordpress.org/ticket/44349.
+				 */
+				$orderby = "{$wpdb->posts}.post_date " . $query_vars['order'] . ', ' . "{$wpdb->posts}.ID " . $query_vars['order'];
 			}
-		} elseif ( 'none' === $query_vars['orderby'] ) {
+			// See get_pages(): when sort_column is 'none', the get_pages() function should not generate any ORDER BY clause.
+			// Should it rather be handled in the get_pages() function?
+			// src/wp-includes/post.php L6496
+		} elseif ( 'none' === $query_vars['orderby'] || ( is_array( $query_vars['orderby'] ) && array_key_exists( 'none', $query_vars['orderby'] ) ) ) {
```

Comment:
Just a nitpick :) Thought I saw somewhere that `isset( array['key'] )` is tiny bit faster/preferable to `is_array()` followed by `array_key_exists()`. Maybe the `( is_array( $query_vars['orderby'] ) && array_key_exists( 'none', $query_vars['orderby'] ) )` could be `isset( $query_vars['orderby']['none'] )`?